packages/flutter/lib/src/material/dialog.dart

```diff
@@ -481,12 +486,15 @@ class _DialogRoute<T> extends PopupRoute<T> {
 Future<T> showDialog<T>({
   @required BuildContext context,
   bool barrierDismissible: true,
-  @required Widget child,
+  @deprecated Widget child,
+  @required WidgetBuilder builder,
 }) {
   return Navigator.of(context, rootNavigator: true).push(new _DialogRoute<T>(
-    child: child,
+    child: child == null ? new Builder(builder: builder) : child,
     theme: Theme.of(context, shadowThemeOnly: true),
     barrierDismissible: barrierDismissible,
-    barrierLabel: MaterialLocalizations.of(context).modalBarrierDismissLabel,
+    barrierLabel: MaterialLocalizations
+        .of(context)
+        .modalBarrierDismissLabel,
```

Comment:
we would usually just put this on one line

Reply:
Done